### DISCUSSION TITLE
How to create a new folder on a repository?<title>

### DISCUSSION THREAD

**Discussion by ERNESTOGith** [community/community#69927] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello everyone,

When I want to create a new folder on a repository, there is not a part on which it is mentionned " commit new file". I don't know why .
Is there somebody to help me?

**Comment by thevindu-w** [comment, marked as answer]
You can create a folder by creating a new file inside it. For example, if you need to create a folder `my_folder`, you can create a new file named `my_folder/.keep` and commit. This will create a folder named `my_folder`. Git is not adding empty directories. It only tracks files. So you need at least one file in the directory.

**Reply by ERNESTOGith** [reply]
Thanks for your help.

**Reply by MADHACKER85** [reply]
thanks bro

**Reply by DylanPrinsloo** [reply]
Nice

**Reply by Bakal47** [reply]
Great! thanks

**Comment by RomanchenkoAS** [comment]
You need to create a directory locally on your pc, add a new file to that directory and push that file to github.

**Reply by ERNESTOGith** [reply]
Thanks for your help

**Comment by igorskyflyer** [comment]
If you want to create an empty folder here on GitHub.com you need to create an empty file inside of it since it doesn't support creation of empty folders.

Here's how:
- Go to your repo,
- Click on "Create new file",
- In the "Name your file" input box type the folder name + "/" (slash symbol) + the new filename (anything works, like `dummy`, `keep`, etc.), e.g.

---

Name your file: new-folder/dummy

---

And voila! 🪄 Your folder `new-folder` will get created!

**Reply by ERNESTOGith** [reply]
Thanks for your help.

**Reply by igorskyflyer** [reply]
No problem, Ernesto, if you have any more issues or need help, we'll be happy to help.

If this issue is resolved, please consider marking a reply as the correct answer. Thanks.

**Comment by ERNESTOGith** [comment]
Thanks for your help.

**Comment by wnqqnw19** [comment]
When creating a new folder on GitHub, ensure you name the folder followed by a file (e.g., new-folder/README.md) and add content to the file. Scroll down to the "Commit new file" section, add a commit message, and click "Commit new file." 
If you don't see this option, check your repository permissions or try a different browser.
